### PR TITLE
tstime/mono: remove unsafe

### PR DESCRIPTION
**Pull request by bradfitz**
This removes the unsafe/linkname and only uses the standard library.

It's a bit slower, for now, but https://go.dev/cl/518336 should get us
back.

On darwin/arm64, without https://go.dev/cl/518336

    pkg: tailscale.com/tstime/mono
              │   before    │                after                │
              │   sec/op    │   sec/op     vs base                │
    MonoNow-8   16.20n ± 0%   19.75n ± 0%  +21.92% (p=0.000 n=10)
    TimeNow-8   39.46n ± 0%   39.40n ± 0%   -0.16% (p=0.002 n=10)
    geomean     25.28n        27.89n       +10.33%

And with it,

    MonoNow-8   16.34n ±  1%   16.93n ± 0%  +3.67% (p=0.001 n=10)
    TimeNow-8   39.55n ± 15%   38.46n ± 1%  -2.76% (p=0.000 n=10)
    geomean     25.42n         25.52n       +0.41%

Updates https://github.com/tailscale/tailscale/issues/8839
Updates https://github.com/tailscale/go/issues/70

### CODE REVIEW

File: cmd/derper/depaware.txt [hunk 1]
```diff
@@ -116,7 +116,7 @@ tailscale.com/cmd/derper dependencies: (generated by github.com/tailscale/depawa
         tailscale.com/tka                                            from tailscale.com/client/tailscale+
    W    tailscale.com/tsconst                                        from tailscale.com/net/interfaces
         tailscale.com/tstime                                         from tailscale.com/derp+
-     💣 tailscale.com/tstime/mono                                    from tailscale.com/tstime/rate
```
Comment (dsnet) [inline cmd/derper/depaware.txt:119]: beautiful 🥇

**Review by dsnet — APPROVED** [review]

**Review by crawshaw — APPROVED** [review]
I believe those nanoseconds matter to our throughput, should we patch that CL into our go branch?

**Comment by bradfitz** [comment]
Sure, I'll patch that in to our tree. With that in, it's only:

```
goos: darwin
goarch: arm64
pkg: tailscale.com/tstime/mono
          │    before    │               after                │
          │    sec/op    │   sec/op     vs base               │
MonoNow-8   16.34n ±  1%   16.93n ± 0%  +3.67% (p=0.001 n=10)
TimeNow-8   39.55n ± 15%   38.46n ± 1%  -2.76% (p=0.000 n=10)
geomean     25.42n         25.52n       +0.41%
```

File: go.toolchain.rev [hunk 2]
```diff
@@ -1 +1 @@
-593313088f6e5297cba5d5e3f392f15251bd5915
```
Comment (bradfitz) [inline go.toolchain.rev:1]: Done: https://github.com/tailscale/go/compare/593313088f6e5297cba5d5e3f392f15251bd5915..27f103a44f8fd34a2cc36995ce7bf83d04433ead

**Review by maisem — APPROVED** [review]

**Pull request merged**